src/abstractdatatable/join.jl

```diff
@@ -196,14 +196,14 @@ function compose_joined_table(joiner::DataTableJoiner,
     dtr_noon = without(joiner.dtr, joiner.on_cols)
     # permutation to swap rightonly and leftonly rows
     right_perm = vcat(1:length(right_ixs),
```

Comment:
Like with the left_ixs suggestion above: we can first allocate `all_orig_right_ixs`, then assign to its parts. To get rid of `right_perm` we have to fill the indices of the left-only rows with some random indices (the corner case is an empty right table).